src/main/java/mate/academy/rickandmorty/service/CharacterServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.rickandmorty.service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Random;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.client.RickAndMortyCharactersClient;
+import mate.academy.rickandmorty.dto.external.ExternalDto;
+import mate.academy.rickandmorty.dto.internal.CharacterDto;
+import mate.academy.rickandmorty.mappper.CharacterMapper;
+import mate.academy.rickandmorty.repository.CharacterDao;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CharacterServiceImpl implements CharacterService {
+    public static final String ENDPOINT = "/api/character";
+    private final CharacterDao characterDao;
+    private final CharacterMapper characterMapper;
+    private final RickAndMortyCharactersClient charactersClient;
+    private final Random random;
+    @Value("${rickandmorty.url}")
+    private String baseUrl;
+
+    @EventListener(ContextRefreshedEvent.class)
```

Comment:
https://stackoverflow.com/a/67227144/13842504
```suggestion
    @EventListener(ApplicationReadyEvent.class)
```